### PR TITLE
100% test coverage for input and output managers.

### PR DESCRIPTION
**Pull request by michaelRichards99**
This PR brings the test coverage for `input_manager.py` and `output_manager.py` as well as their respective unit test files to `100%`.

Very slight modifications were made to the source files themselves in order to remove unreachable code / conditions and to remove occurrences of a single `\n` byte.

### CODE REVIEW

File: RUFAS/input_manager.py [hunk 1]
```diff
@@ -258,11 +262,7 @@ def _validate_element(self, element_hierarchy: List[str], properties_blob_key: s
             return element_counter_and_validity
         else:
             var_name = element_hierarchy[-1]
-            try:
-                input_data_value = reduce(lambda d, key: d[key], element_hierarchy, input_data)
-            except KeyError:
-                raise KeyError(f"Key {var_name} not found in input data")
```
Comment (michaelRichards99) [inline RUFAS/input_manager.py:264]: This exception case was unreachable due to the same `reduce()` being called above.

File: RUFAS/input_manager.py [hunk 2]
```diff
@@ -229,8 +229,12 @@ def _validate_element(self, element_hierarchy: List[str], properties_blob_key: s
                     }
         element_counter_and_validity = {"fixed_elements": 0, "total_elements": 0, "valid_elements": 0,
                                         "invalid_elements": 0, "is_valid": True}
-        variable_properties = reduce(lambda d, key: d[key], element_hierarchy,
-                                     self.__metadata["properties"][properties_blob_key])
+        try:
+            variable_properties = reduce(lambda d, key: d[key], element_hierarchy,
+                                         self.__metadata["properties"][properties_blob_key])
+        except KeyError as e:
+            raise e
```
Comment (Isgotkowitz) [inline RUFAS/input_manager.py]: Would it be better/is it possible to display a more detailed error message here?

**Review by Isgotkowitz — APPROVED** [review]
Nice PR Michael, only comment is about possibly raising a more detailed error message but is up to you. Also have pulled this code to local and can confirm that everything passes on Windows

**Review by allisterakun — APPROVED** [review]
All the changes look good to me! 
I reviewed the IM-related changes, great changes and additions. I am not as familiar with the OM, but I read through the changes and they seem sensible to me.
Great PR!

File: RUFAS/input_manager.py (hunk 2 above)
Comment (michaelRichards99) [reply RUFAS/input_manager.py]: Done.

**Comment by michaelRichards99** [comment]
> Nice PR Michael, only comment is about possibly raising a more detailed error message but is up to you. Also have pulled this code to local and can confirm that everything passes on Windows

Thanks for checking on windows.

**Pull request merged**